### ISSUE TITLE
Not able to create key successfully

### ISSUE THREAD

**Issue by tomasfrancisco** [integrated-reasoning/KeyHippo#16]
```typescript
const keyHippo = new KeyHippo(ctx.supabase);
const newApiKey = await keyHippo.createApiKey(
  ctx.account.userId,
  input.description
);

console.log(newApiKey); // ["{\"id\": \"e6b448a4-e652-4180-8aaa-5ab8d38d2a9d\", \"description\": \"fc09615c-f6e4-40e7-87cb-a26702c5d8bd-3af1c6f4-3fac-49a1-b43a-f29bb5cf5248-New\"}"]
```

```
API key info loaded for user: d300e31a-38fb-4f66-a408-4d33cc144f99. Count: 1
Loaded key info: [{}]
Failed to create new API key: Failed to find the newly created API key

...

Error creating API key: Failed to find the newly created API key
```

If I try to load the keys, this is the kind of data I get:

```json
[
  {
    "name": "fc09615c-f6e4-40e7-87cb-a26702c5d8bd-3af1c6f4-3fac-49a1-b43a-f29bb5cf5248-New",
    "permission": "readOnly",
    "last_used": null,
    "created": "2024-09-02T16:18:35.342534+00:00",
    "revoked": null,
    "total_uses": 0,
    "success_rate": 100,
    "total_cost": 0
  }
]

```

But I'm unable to acquire the key itself to be used for authentication.

Am I missing something?

**Comment by david-r-cox** [comment]
Hey @tomasfrancisco, sorry you're hitting that and for the lack of documentation (it's coming soon!).

The key should be returned by `createApiKey` in the `apiKey` field. From your logs, it looks like a key was created but not returned. I'm aware of a few cases in which this can happen, the most common being generating a key for a user that existed prior to the installation of KeyHippo.

I'd be glad to get you setup on a quick call. KeyHippo still has some rough edges, mostly during the onboarding process. I'll be working to address them over the next few weeks.

Also regarding loading the keys: `loadApiKeyInfo` and `getAllKeyMetadata` don't return the key itself -- it's only visible once in the return value of `createApiKey` and can't be re-derived again. I updated the client's readme to hopefully clarify the usage of those (admittedly poorly named) functions: https://github.com/integrated-reasoning/KeyHippo/pull/17

Thanks for opening the issue and trying KeyHippo!

**Comment by david-r-cox** [comment]
@tomasfrancisco I believe this is now fixed as of [v0.0.31](https://github.com/integrated-reasoning/KeyHippo/releases/tag/v0.0.31).
The fix was introduced in [v0.0.26](https://github.com/integrated-reasoning/KeyHippo/releases/tag/v0.0.31) but it was implemented as an update rather than a self-contained installation which ended up breaking our dbdev package (see https://github.com/integrated-reasoning/KeyHippo/pull/20#issuecomment-2327639681). I've reverted to self-contained installations for now as KeyHippo is still in alpha.

Mind giving it another go with v0.0.31?

**Comment by tomasfrancisco** [comment]
Hey @david-r-cox, thanks a lot for the quick resolution!
I tried it out and I now get this:

```sh
API key info loaded for user: 9e79ad59-0bb9-4542-9283-e95b64a99b7e. Count: 1
Loaded key info: [{"id":"a5837cdc-d761-4601-80cd-8c33759084a3","description":"c6ed44d0-e2cf-4ab2-b40e-0d3dfb5fcb32-481d6438-f714-436a-aaf5-907c815a8e12-Hello"}]
New API key created for user: 9e79ad59-0bb9-4542-9283-e95b64a99b7e, Key ID: a5837cdc-d761-4601-80cd-8c33759084a3
{
  newApiKey: {
    id: 'a5837cdc-d761-4601-80cd-8c33759084a3',
    description: 'c6ed44d0-e2cf-4ab2-b40e-0d3dfb5fcb32-481d6438-f714-436a-aaf5-907c815a8e12-Hello',
    apiKey: '769b87c8106d57fccd085d4e3d0fbc677e02838f42f8c497b1cd23720cc76bc05e18b2799d5d720a75577f0232072a09d0b93bdf702240982c55fffb881a293b',
    status: 'success'
  }
}
```

I think it's safe to say: this issue can be closed.
